TACHO_DRIVER.md

```diff
@@ -0,0 +1,199 @@
+| `on`               | Return EPERM          | Works
+
+The `ramp_up_sp` and `ramp_down_sp` specify how long (in msec) the motor
+will take to go from 0 to 100 of the controlling parameter depending on
+the `speed_regulation` mode.
+
+| `speed_regulation` | Ramp control parameter
+| ------------------ | ----------------------
+| `off`              | 0 to 100% Duty Cycle                 
+| `on`               | 0 to 100% Max Speed
+
+The maximum speed of the motor is set in the driver. The large motor has a
+maximum speed of 1200 pulses per second and the medium motor has a maximum
+speed of 900 pulses per second. Both motors have 360 pulses per revolution.
+
+## Ramping
+
+See `spec.json` for a description of the `ramp_up_sp` and `ramp_down_sp`
+attributes. Except in `run-direct` mode, you cannot change the ramping
+attributes while the motor is running.
+
+If you change the allowed `duty_cycle_sp` or `speed_sp` value while the
+motor is in `run_direct` mode, the ramp time setpoints govern the operation.
+
+No matter the sign of the `duty_cycle_sp` or `speed_sp`, if the new
+setpoint is towards a lower absolute value, then the `ramp_down_sp`
+determines how the motor slows down. If the new setpoint is towards a
+higher absolute value, then the `ramp_up_sp` determines how the motor
+speeds up.
+
+An example follows:
+
+- `ramp_up_sp` is set to 4000
+- `ramp_down_sp` is set to 2000
+- `speed_regulation` is set to `off`
+- `duty_cycle_sp` is set to 75%
+
+If we write `run-direct` to the motor `command` attribute, the motor will
+take 3000 msec to ramp up to 75% duty cycle. What happens when we change the
+`duty_cycle_sp` to other values while the motor is running?
+
+| New `duty_cycle_sp` | Operation
+| ------------------- | ----------------------
+|   100               | Motor ramps up to 100% in 1000 msec
+|   50                | Motor ramps down to 50% in 1000 msec
+|   -50               | Motor ramps down to 0% in 1000 msec and up to -50 in 2000 msec
+
+## Run Modes
+
+The `run` attribute determines function that the motor is executing. Within a `run`
+mode, some attributes have an effect, while others do not. These details are
+described below.
+
+The run modes can be classified into two types. The `run-direct` is suited to
+applications where the programmer wants direct control of the motors. The other
+three modes:
+
+- `run_timed`
+- `run_to_abs_pos`
+- `run_to_rel_pos`
+
+They can be through of as convenience commands where the driver handles the motor
+control independently of the application. This is well suited to applications where you
+want the driver to handle the details, like machines where the motors must alternate
+between precise positions or run for fixed timing periods.
+
+### `run-direct` Mode
+
+In its most basic operating mode, the `tacho-motor` accepts the `run-direct`
+command. After accepting a `run-direct` command, the motor will only accept
+a `stop` command - the other commands represent "helpers" that automatically
```

Comment:
The other commands are intended to implement all-in-one operations. I think we should keep them separate form `run-direct`.

On the other hand, I can see a case for allowing `run-timed` when in `run-direct` mode to allow the motor to run for `n` more msecs, or something like that.